client/client_test.go

```diff
@@ -1370,64 +1351,117 @@ func testPublishNoOneHasSnapshotKey(t *testing.T, rootType string) {
+
+	repo, _ = initializeRepo(t, data.ECDSAKey, "docker.com/notary2", ts.URL, false)
+	defer os.RemoveAll(repo.baseDir)
+	testPublishBadMetadata(t, data.CanonicalSnapshotRole, repo, false, false)
+
+	// publish first - should succeed
+	repo, _ = initializeRepo(t, data.ECDSAKey, "docker.com/notary3", ts.URL, true)
+	defer os.RemoveAll(repo.baseDir)
+	testPublishBadMetadata(t, data.CanonicalSnapshotRole, repo, true, true)
+
+	repo, _ = initializeRepo(t, data.ECDSAKey, "docker.com/notary4", ts.URL, false)
+	defer os.RemoveAll(repo.baseDir)
+	testPublishBadMetadata(t, data.CanonicalSnapshotRole, repo, true, true)
 }
 
-func testPublishBadExistingSnapshot(t *testing.T, rootType string,
-	serverManagesSnapshot bool, readable bool) {
+// If the targets metadata is corrupt or the targets metadata is unreadable,
+// we can't publish for the first time, because there is no existing data for.
+// us to download. If the repo has already been published, it doesn't matter
+// if the metadata is corrupt because we can just redownload if it is.
+func TestPublishTargetsCorrupt(t *testing.T) {
+	ts := fullTestServer(t)
+	defer ts.Close()
 
-	// Temporary directory where test files will be created
-	tempBaseDir, err := ioutil.TempDir("/tmp", "notary-test-")
-	defer os.RemoveAll(tempBaseDir)
-	assert.NoError(t, err, "failed to create a temporary directory: %s", err)
+	// do not publish first - publish should fail
+	repo, _ := initializeRepo(t, data.ECDSAKey, "docker.com/notary1", ts.URL, false)
+	defer os.RemoveAll(repo.baseDir)
+	testPublishBadMetadata(t, data.CanonicalTargetsRole, repo, false, false)
+
+	// publish first - should succeed
+	repo, _ = initializeRepo(t, data.ECDSAKey, "docker.com/notary2", ts.URL, false)
+	defer os.RemoveAll(repo.baseDir)
+	testPublishBadMetadata(t, data.CanonicalTargetsRole, repo, true, true)
+}
+
+// If the root metadata is corrupt or the root metadata is unreadable,
+// we can't publish for the first time or the second time.  Root is the most
+// important and what we used to pin trust, so if it's corrupt, we can't
+// verify downloaded updates.
+func TestPublishRootCorrupt(t *testing.T) {
+	t.Skip("Test currently fails - not sure what the correct behavior is.")
 
-	gun := "docker.com/notary"
 	ts := fullTestServer(t)
 	defer ts.Close()
 
-	repo, _ := initializeRepo(
-		t, rootType, tempBaseDir, gun, ts.URL, serverManagesSnapshot)
+	// do not publish first - publish should fail
+	repo, _ := initializeRepo(t, data.ECDSAKey, "docker.com/notary1", ts.URL, false)
+	defer os.RemoveAll(repo.baseDir)
+	testPublishBadMetadata(t, data.CanonicalRootRole, repo, false, false)
+
+	// publish first - publish should still succeed if root corrupt
+	repo, _ = initializeRepo(t, data.ECDSAKey, "docker.com/notary2", ts.URL, false)
+	defer os.RemoveAll(repo.baseDir)
+	testPublishBadMetadata(t, data.CanonicalRootRole, repo, true, false)
+}
+
+// When publishing snapshot, root, or target, if the repo hasn't been published
```

Comment:
Yes, if the repo hasn't been published yet, corrupt root data will cause a publish to fail.  If it *has* been published, it seems to be fine-ish because we re-download.  (there's a weird error when I can't read the file - it's `ErrLocalRootExpired` instead of a path error, which I expected, but I wanted to make sure that it *should* be fine, and not just blow up)